compiler/damlc/tests/src/DA/Test/Packaging.hs

```diff
@@ -431,6 +431,152 @@ tests damlc repl davlDar = testGroup "Packaging" $
+          createDirectoryIfMissing True (projB </> "src")
+          createDirectoryIfMissing True (projC </> "src")
+
+          writeFileUTF8 (projA </> "daml.yaml") $ unlines
+            [ "sdk-version: " <> sdkVersion
+            , "name: a"
+            , "version: 0.0.1"
+            , "source: src"
+            , "dependencies: [daml-prim, daml-stdlib]"
+            ]
+          writeFileUTF8 (projA </> "src" </> "A.daml") $ unlines
+            [ "daml 1.2"
+            , "module A where"
+            , "foo : Int"
+            , "foo = 10"
+            ]
+          withCurrentDirectory projA $ callProcessSilent damlc ["build", "-o", "a.dar"]
+          packageIdA1 <- head <$> darPackageIds (projA </> "a.dar")
+
+          writeFileUTF8 (projB </> "daml.yaml") $ unlines
+            [ "sdk-version: " <> sdkVersion
+            , "name: b"
+            , "version: 0.0.1"
+            , "source: src"
+            , "dependencies:"
+            , " - daml-prim"
+            , " - daml-stdlib"
+            , " - " <> projA </> "a.dar"
+            ]
+          writeFileUTF8 (projB </> "src" </> "B.daml") $ unlines
+            [ "daml 1.2"
+            , "module B where"
+            , "import A ()"
+            ]
+          withCurrentDirectory projB $ callProcessSilent damlc ["build", "-o", "b.dar"]
+
+          writeFileUTF8 (projA </> "src" </> "A.daml") $ unlines
+            [ "daml 1.2"
+            , "module A where"
+            , "foo : Int"
+            , "foo = 20"
+            ]
+          withCurrentDirectory projA $ callProcessSilent damlc ["build", "-o", "a.dar"]
+          packageIdA2 <- head <$> darPackageIds (projA </> "a.dar")
+          assertBool "Expected two different package IDs" (packageIdA1 /= packageIdA2)
+
+          writeFileUTF8 (projC </> "daml.yaml") $ unlines
+            [ "sdk-version: " <> sdkVersion
+            , "name: c"
+            , "version: 0.0.1"
+            , "source: src"
+            , "dependencies:"
+            , " - daml-prim"
+            , " - daml-stdlib"
+            , " - " <> projA </> "a.dar"
+            , " - " <> projB </> "b.dar"
+            ]
+          writeFileUTF8 (projC </> "src" </> "C.daml") $ unlines
+            [ "daml 1.2"
+            , "module C where"
+            , "import A ()"
+            , "import B ()"
+            ]
+          buildProjectError projC "" "dependencies with same unit id but conflicting package ids: a-0.0.1"
+
+    , testCase "Detects unitId collisions in data-dependencies" $ withTempDir $ \projDir -> do
+          -- Check that two pacages with the same unit id is flagged as an error.
+          let projA = projDir </> "a"
+          let projB = projDir </> "b"
+          let projC = projDir </> "c"
+
+          createDirectoryIfMissing True (projA </> "src")
+          createDirectoryIfMissing True (projB </> "src")
+          createDirectoryIfMissing True (projC </> "src")
+
+          writeFileUTF8 (projA </> "daml.yaml") $ unlines
```

Comment:
We should really make a helper function for creating `daml.yaml` files in tests :slightly_smiling_face: